src/librustc_trans/trans/cleanup.rs

```diff
@@ -769,20 +771,24 @@ impl<'blk, 'tcx> CleanupHelperMethods<'blk, 'tcx> for FunctionContext<'blk, 'tcx
                 }
             }
 
+            // Pop off the scope, since we may be generating
+            // unwinding code for it.
+            popped_scopes.push(self.pop_scope());
```

Comment:
Nit: `let popped = self.pop_scope()` here, use `popped` in the block below and `popped_scopes.push(popped)` after that block.